include/boost/windows_sspi/detail/sspi_impl.hpp

```diff
@@ -119,11 +135,28 @@ class sspi_decrypt {
         encrypted_data = std::vector<char>(reinterpret_cast<const char*>(pExtraBuffer->pvBuffer), reinterpret_cast<const char*>(pExtraBuffer->pvBuffer) + pExtraBuffer->cbBuffer);
       }
     }
-    return sc;
+    BOOST_ASSERT(!decrypted_data.empty());
+
+    return state::data_available;
+  }
+
+  // TODO: Consider making this more flexible by not requering a
+  // vector of chars, but any view of a range of bytes
+  void put(const std::vector<char>& data) {
+    encrypted_data.insert(encrypted_data.end(), data.begin(), data.end());
+  }
+
+  std::vector<char> get(std::size_t max) {
+    std::size_t size = std::min(max, decrypted_data.size());
+    std::vector<char> ret{decrypted_data.begin(), decrypted_data.begin() + size};
+    decrypted_data.erase(decrypted_data.begin(), decrypted_data.begin() + size);
+    return ret;
   }
 
+  // TODO: Make private
   std::vector<char> encrypted_data;
   std::vector<char> decrypted_data;
+  SECURITY_STATUS error_code;
```

Comment:
naming this variable error_code threw me off since it's the name name as the system::error_code type.

Reply:
Understandable :-) I will renamed it.